lib/fog/azurerm/credentials.rb

```diff
@@ -19,19 +19,26 @@ def self.get_new_credentials(tenant_id, client_id, client_secret)
         @tenant_id = tenant_id
         @client_id = client_id
         @client_secret = client_secret
+        return if @tenant_id.nil? || @client_id.nil? || @client_secret.nil?
         @token_provider = MsRestAzure::ApplicationTokenProvider.new(@tenant_id, @client_id, @client_secret)
         @credentials = MsRest::TokenCredentials.new(@token_provider)
         @credentials
       end
 
-      def self.new_client(tenant_id, client_id, client_secret)
+      def self.new_management_credential?(tenant_id, client_id, client_secret)
         @tenant_id != tenant_id ||
           @client_id != client_id ||
           @client_secret != client_secret
       end
 
+      def self.new_account_credential?(options = {})
+        @account_name != options[:azure_storage_account_name] ||
+          @account_key != options[:azure_storage_access_key] ||
+          @connection_string != options[:azure_storage_connection_string]
```

Comment:
There can be azure_storage_account_name and azure_storage_access_key only. azure_storage_connection_string is not necessary.